pkg/task/task.go

```diff
@@ -69,21 +70,104 @@ func NewTask(raw string) (Task, error) {
 		}
 	default:
 		// throw std::string ("Record not recognized as format 4.");
-		// @TODO parseLegacy
-		return Task{}, fmt.Errorf("record not recognized as format 4")
+		log.Debugf("record not recognized as format 4")
+		return parseLegacy(raw)
 	}
 
 	// recalc_urgency = true;
 
 	return task, nil
 }
 
-func fromJSON(json string) string {
-	// @TODO implement
-	return json
-}
+func parseLegacy(line string) (Task, error) {
+	switch determineVersion(line) {
+	// File format version 1, from 2006-11-27 - 2007-12-31, v0.x+ - v0.9.3
+	case 1:
+		return Task{}, fmt.Errorf("taskwarrior no longer supports file format 1, originally used between 27 November 2006 and 31 December 2007")
+
+	// File format version 2, from 2008-1-1 - 2009-3-23, v0.9.3 - v1.5.0
+	case 2:
+		return Task{}, fmt.Errorf("taskwarrior no longer supports file format 2, originally used between 1 January 2008 and 12 April 2009")
+
+	// File format version 3, from 2009-3-23 - 2009-05-16, v1.6.0 - v1.7.1
+	case 3:
+		return Task{}, fmt.Errorf("taskwarrior no longer supports file format 3, originally used between 23 March 2009 and 16 May 2009")
+
+	// File format version 4, from 2009-05-16 - today, v1.7.1+
+	case 4:
```

Comment:
v4 doesn't make sense when parsing legacy. Return a proper error.  Perhaps this is the default return error at line 106?